src/typechecker/FStar.TypeChecker.TcEffect.fs

```diff
@@ -660,10 +660,11 @@ Errors.with_ctx (BU.format1 "While checking layered effect definition `%s`" (str
      *
      * Don't think the boolean false below matters, but is perhaps safer (see Syntax.fsi)
      *)
-    let binder_aq_to_arg_aq aq =
-      match aq with
-      | Some (Implicit _) -> aq
-      | Some (Meta _) -> Some (Implicit false)
+    let binder_aq_to_arg_aq (aq, attrs) =
+      match aq, attrs with
+      | Some (Implicit _), _ -> aq
+      | Some (Meta _), _
+      | None, _ when List.length attrs <> 0 -> Some (Implicit false)
```

Comment:
Shouldn't this simply return None?

Reply:
Thanks, I just removed this line since it was not being used.